### ISSUE TITLE
`sct_process_segmentation` always uses the `'Slice (I->S)'` column regardless of the input orientation  

### ISSUE THREAD

**Issue by valosekj** [spinalcordtoolbox/spinalcordtoolbox#4363]
### Description

The [compute_shape()](https://github.com/spinalcordtoolbox/spinalcordtoolbox/blob/764aa9a3313b5d949b5117ed05b3b84a06737066/spinalcordtoolbox/process_seg.py#L25) function used by the `sct_process_segmentation` script reorient an input segmentation to RPI (Right-Posterior-Inferior) before computing shape metrics:

https://github.com/spinalcordtoolbox/spinalcordtoolbox/blob/764aa9a3313b5d949b5117ed05b3b84a06737066/spinalcordtoolbox/scripts/sct_process_segmentation.py#L423

https://github.com/spinalcordtoolbox/spinalcordtoolbox/blob/764aa9a3313b5d949b5117ed05b3b84a06737066/spinalcordtoolbox/process_seg.py#L50

The shape metrics are then outputted into a CSV file with a predefined header. One of the header columns is `'Slice (I->S)'`:

https://github.com/spinalcordtoolbox/spinalcordtoolbox/blob/764aa9a3313b5d949b5117ed05b3b84a06737066/spinalcordtoolbox/aggregate_slicewise.py#L559

_**In other words, regardless of the input image orientation, the CSV file header is always the same, containing `'Slice (I->S)'`.**_

This might introduce confusion when working with images with "S orientation", such as ASL (Anterior-Superior-Left):

```console
$ sct_image -i sub-01_T2w.nii.gz -setorient ASL -o sub-01_T2w_ASL.nii.gz
$ sct_image -i sub-01_T2w_ASL.nii.gz -header | grep -E qform_[xyz]
qform_xorient	Anterior-to-Posterior
qform_yorient	Superior-to-Inferior
qform_zorient	Left-to-Right
```

Because the CSV output of `sct_process_segmentation` for the ASL-oriented image still contains slices for RPI orientation:

```console
$ sct_process_segmentation -i sub-01_T2w_seg_ASL.nii.gz -vert 2:3 -vertfile sub-01_T2w_seg_labeled_ASL.nii.gz
```

<img width="600" alt="image" src="https://github.com/spinalcordtoolbox/spinalcordtoolbox/assets/39456460/95527999-60f9-4544-9f6a-33e52979fc90">

This means that, for example, slice 199 corresponding to the rostral slice of the C2 level in "I orientation" does not correspond to C2 anymore for "S orientation":

<img width="300" alt="AIL" src="https://github.com/spinalcordtoolbox/spinalcordtoolbox/assets/39456460/4cb106ec-b886-4510-ad50-fef3bc084d2e">
<img width="300" alt="ASL" src="https://github.com/spinalcordtoolbox/spinalcordtoolbox/assets/39456460/27f7a4ef-e1a3-4563-8000-9cc01d581c67">

**Comment by joshuacwnewton** [comment]
Excellent investigation! 

tl;dr: The CSV is technically accurate, but the specific slice number can't be copied and pasted directly into FSLeyes if the image is flipped in the IS axis.

In that case, I have some thoughts:

- How common is the 'S' orientation? I feel like in practice, I've mainly seen images in AIL, RPI, and LPI orientation. So, maybe this isn't actually a problem in 99% of cases?
- Could we make this clearer for the user by renaming `'Slice (I->S)'` to be `'IS Slice (I = 0)'`? (Indicating that IS slice number is only valid for images where 0 = the inferior-most slice.)
- Could we eliminate the confusion entirely by having two columns? One for IS slice # (`'IS slice (I = 0)'`), and one for SI slice # (`'SI slice (S = 0)'`)? They'd technically be redundant / inverses of one another, but it would at least cover all image orientations :P

**Comment by mguaypaq** [comment]
> * Could we eliminate the confusion entirely by having two columns? One for IS slice # (`'IS slice (0 = I)'`), and one for SI slice # (`'SI slice (0 = S)'`)? They'd technically be redundant / inverses of one another, but it would at least cover all image orientations :P

Personally, I like this suggestion; it seems to me like this would be pretty user-friendly.

**Comment by valosekj** [comment]
> The CSV is technically accurate

Yes!

> * How common is the 'S' orientation? I feel like in practice, I've mainly seen images in AIL, RPI, and LPI orientation. So, maybe this isn't actually a problem in 99% of cases?

I have a similar feeling.

---

Notice that slice 199 corresponds to the rostral slice of the C2 level in "I orientation". BUT, for "S orientation", slice 199 corresponds to C5, i.e., the completely wrong level.
